packages/block-editor/src/components/block-tools/block-contextual-toolbar.js

```diff
@@ -110,15 +110,18 @@ function BlockContextualToolbar( { focusOnMount, isFixed, ...props } ) {
 		}, [] );
 
 	useEffect( () => {
-		setIsCollapsed( false );
+		if ( ! selectedBlockClientId ) {
+			setIsCollapsed( false );
+		}
 	}, [ selectedBlockClientId ] );
 
 	const isLargeViewport = useViewportMatch( 'medium' );
 
-	if ( blockType ) {
-		if ( ! hasBlockSupport( blockType, '__experimentalToolbar', true ) ) {
-			return null;
-		}
+	if (
+		blockType &&
+		! hasBlockSupport( blockType, '__experimentalToolbar', true )
+	) {
+		return null;
```

Comment:
This is just a tiny code quality improvement; I fixed it since I was in the "area."